configextractor.py

```diff
@@ -89,60 +137,31 @@ def section_builder(self, parser, field_dict, result, parsertype="MWCP"):
             parser_section.set_heuristic(HEURISTICS_MAP.get(category, 1), attack_id=mitre_att)
             parser_section.add_tag("source", parsertype)
-            parser_section.add_tag('attribution.implant', malware_name.upper())
-            if mitre_group :
+            if malware_name:
+                parser_section.add_tag('attribution.implant', malware_name.upper())
+            if mitre_group:
                 parser_section.add_tag('attribution.actor', mitre_group.upper())
             for malware_type in malware_types:
                 parser_section.add_tag('attribution.family', malware_type.upper())
-        fields_dictstrings = {"other": "file.config"}
-        fields_liststringtuples = {"port": "network.dynamic.port", "socketaddress": "", "c2_socketaddress": "",
-                                   "credential": "", "ftp": "", "listenport": "", "outputfile": "", "proxy": "",
-                                   "proxy_socketaddress": "", "registrypathdata": "", "rsa_private_key": "",
-                                   "rsa_public_key": "", "service": ""}
-        for field, tag in fields_liststrings.items():
-            if field in field_dict:
-                generic_section = ResultSection(f" {field.capitalize()}  ")
-                for line in field_dict[field]:
-                    generic_section.add_line(f"{line}")
-                if tag:
-                    for x in field_dict[field]:
-                        generic_section.add_tag(tag, x)
-                parser_section.add_subsection(generic_section)
-
-        for field, tag in fields_dictstrings.items():
-            if field in field_dict:
-                other_section = ResultSection(f"Other metadata found ", body_format=BODY_FORMAT.KEY_VALUE,
-                                              body=json.dumps(field_dict["other"]))
-
-                parser_section.add_subsection(other_section)
-
-        for field, tag in fields_liststringtuples.items():
-            if field in field_dict:
-                generic_section = ResultSection(f"{field.capitalize()} Section ")
-                for lst in field_dict[field]:
-                    generic_section.add_lines(lst)
-                if tag:
-                    for x in field_dict[field]:
-                        generic_section.add_tag(tag, x)
-                parser_section.add_subsection(generic_section)
+        # Create subsections and attach them to the main parser_section
+        subsection_builder(parser_section, field_dict)
+
+        other_key = "other"
+        if other_key in field_dict:
+            other_content = field_dict[other_key]
+            other_section = ResultSection(f"Other metadata found", body_format=BODY_FORMAT.KEY_VALUE,
+                                          body=json.dumps(other_content))
+            for field in other_content:
+                other_section.add_tag("file.config", field)
```

Comment:
not sure how I feel about tagging all the different other metadata fields. @cccs-jp  thoughts?
![image](https://user-images.githubusercontent.com/62077906/98415190-cc2b7180-204a-11eb-9d3d-5afc40a1bbe8.png)

Reply:
@cccs-aa is this output from mwcp, or RatDecoder.  If from mwcp I would have expected to get domain:port  as port by themselves are not as useful. I see little value in tagging just port by themselves.

Reply:
@cccs-aa Oh, I saw this [mapping ](https://github.com/CybercentreCanada/assemblyline-service-configextractor/blob/220a0409d6823cc46c976aa4458a2cfd4e646cf6/configextractor.py#L89) between "other" and "file.config" and then saw that the tag wasn't being used [here ](https://github.com/CybercentreCanada/assemblyline-service-configextractor/blob/220a0409d6823cc46c976aa4458a2cfd4e646cf6/configextractor.py#L104) for the section so I figured I would add it. Not sure what the goal of the "other" section is in terms of tagging.

Reply:
That is output from Ratdecoder, agree on getting the domain:port. other is just a section to where everything that doesn't fit in a predefined field goes.

Reply:
Upon discussion, remove "other":"file.config" map due to irrelevancy